open-authoring/open-authoring/OpenAuthoringProvider.tsx

```diff
@@ -57,11 +57,26 @@ export const OpenAuthoringProvider = ({
   const [authorizing, setAuthorizing] = useState(false)
 
   const updateAuthChecks = async () => {
-    setAuthenticated(!!(await cms.api.github.getUser()))
-    setForkValid(await cms.api.github.getBranch(getForkName(), getHeadBranch()))
+    const user = await cms.api.github.getUser()
+    if (user) {
+      setAuthenticated(true)
+      const forkName = getForkName() || (user.login || "") + "/" + process.env.REPO_FULL_NAME.split("/")[1]
```

Comment:
I'm a bit torn on this optimization. It increases the complexity of this function a bit, and I'm not sure that it will scale once we add more use-cases (forking to an organization, accessing an other user's fork, etc)